### ISSUE TITLE
Azure - Resize App service plan - B SKU absent

### ISSUE THREAD

**Issue by justinhauer** [cloud-custodian/cloud-custodian#3140]
### Description: 
The Basic (B) tier SKU is absent from the code in appserviceplan.py which was merged in with #3130. Output from a policy run of trying to resize an app service plan currently in B tier results with an error. Is it not possible to resize a B tier? I'm unclear if the tier was intentionally left out of the recently merged code: 

### Output
```
  File "/virtualenv/lib/python3.7/site-packages/c7n_azure/resources/appserviceplan.py", line 89, in process_resource_set
    client.app_service_plans.update(plan['resourceGroup'], plan['name'], model)
  File "/virtualenv/lib/python3.7/site-packages/azure/mgmt/web/operations/app_service_plans_operations.py", line 465, in update
    raise models.DefaultErrorResponseException(self._deserialize, response)
azure.mgmt.web.models.default_error_response_py3.DefaultErrorResponseException: Operation returned an invalid status code 'Bad Request'
```
Adding in the B tier to the static method and the schema in my pip env still results with the same error: 
```    @staticmethod
 92     def lookup_tier(size):
 93         tiers = {
 94             'F': 'Free',
 95             'B': 'Basic',
 95             'S': 'Standard',
 96             'D': 'Shared',
 97             'P': 'Premium'
 98         }
```
```    schema = utils.type_schema(
 64         'resize-plan',
 65         **{
 66             'size': {'type': 'string', 'enum': ['F1', 'B1', 'B2', 'B3', 'D1', 'S1', 'S2', 'S3', 'P1', 'P2', 'P3']},
 67             'count': {'type': 'integer'}
 68         }
 69     )
```

**Comment by stefangordon** [comment]
I updated that PR with an exception handler that dumps the inner exception object out before the exception is raised.  That is the only way to really see what is causing the "bad request" -- If you are updating that code by hand anyway perhaps add that exception handler in and see what is behind your error? 

https://github.com/capitalone/cloud-custodian/pull/3141/files#diff-427f53eef9225c43717c50667f730a4dR90

**Comment by jdogg89** [comment]
@stefangordon thanks I'll start testing this morning

**Comment by stefangordon** [comment]
fwiw my policy looked like this

```
policies:
  - name: azure-resize-plan
    resource: azure.appserviceplan
    filters:
     - type: 'value'
       key: 'name'
       op: 'eq'
       value_type: 'normalize'
       value: 'test-appserviceplan'
    actions:
     - type: resize-plan
       size: 'B1'
```

**Comment by jdogg89** [comment]
Perhaps the issue is the policy I'm using, I'll try yours out and see what my results are. Here's the policy I'm using:
```
policies:
  - name: azure-resize-plan
    resource: azure.appserviceplan
    filters:
      - tag:scaledown: present
    actions:
     - type: resize-plan
       size: 'S2'
```
Here's the output after adding in your try/catch. It doesn't like the sku.name for some reason:

```
2018-11-15 07:58:20,970: custodian.actions:ERROR Failed to resize cloud-custodian.  Inner exception: {'additional_properties': {'Details': [{'Message': 'The parameter SKU.Name has an invalid value.'}, {'Code': 'BadRequest'}, {'ErrorEntity': {'MessageTemplate': 'The parameter {0} has an invalid value.', 'Parameters': ['SKU.Name'], 'Code': 'BadRequest', 'Message': 'The parameter SKU.Name has an invalid value.'}}], 'Code': 'BadRequest', 'Target': None, 'Innererror': None, 'Message': 'The parameter SKU.Name has an invalid value.'}, 'error': None}
2018-11-15 07:58:20,970: custodian.actions:ERROR Execution failed with error: Operation returned an invalid status code 'Bad Request'
```
When I checkout the resources.json file it looks like the sku name shows up as B1: 
```
 },
 "properties": {
"status": "Ready",
"maximumNumberOfWorkers": 3,
 "numberOfSites": 0,
"resourceGroup": "cloud-custodian"
 },
"sku": {
 "name": "B1",
"tier": "Basic",
 "size": "B1",
"family": "B",
"capacity": 1
},
"resourceGroup": "cloud-custodian",
"c7n:MatchedFilters": [
"tag:scaledown"
```

**Comment by jdogg89** [comment]
It appears that my env matches yours, and yes I am able to switch the plan size within the portal, I tried several times successfully:
```
(pypiCustodian) ➜  custodianDocker pip show azure.mgmt.web
Name: azure-mgmt-web
Version: 0.40.0
Summary: Microsoft Azure Web Apps Management Client Library for Python
```
I've been testing on app service plans just created from the portal, and the app service plan that is created when you go to create an event grid function (if one is not present or specified). I'll try out your ARM template and test the policy out on the App service plan that comes from that.

**Comment by jdogg89** [comment]
I'm not sure what impact if anything it would have but I've been testing in a virtualenv on my mac, not on the linux docker image I have running in AKS that is running other policies for me.

**Comment by jdogg89** [comment]
@stefangordon Using your policy with your ARM template works, I am successfully scaling App service plans! I find it really odd though that app service plans that are created in the GUI, or the app service plan that's created by custodian (in my previous note) when you create event grid functions, is not able to scale - all the previous errors I've mentioned appear when trying that. So, it appears that how  you create an app service plan affects if it is able to be scaled or not....

**Comment by jdogg89** [comment]
@stefangordon I think this issue can be closed if you're ok with the fact that we can just get it to work with an ARM template deployment of the ASP. If you're able to make it work with a deployment from the portal or another way and make it work, we can chalk up the non working ways to some quirk on my end.

**Comment by stefangordon** [comment]
I'm suspicious the app plans might behave different depending if the flavor is marked Function vs. App (although they are basically the same).  I'll play with some function ones and see if I get the same issue.

**Comment by stefangordon** [comment]
The only place I have ever seen this done before is in the Azure CLI - they do it almost the same but they only set `tier` and `name` and not the other two fields.  I'm going to mimic what they have, perhaps the extra fields are causing compatibility issues.

https://github.com/Azure/azure-cli/blob/d1d2b991d2a8a69261e468188be93c6fb2c01884/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py#L1095

**Comment by jdogg89** [comment]
@stefangordon any consensus on the app plans after a few days?

**Comment by stefangordon** [comment]
Oh I will try to get that pending PR in today to lock in with the way the CLI does this.

**Comment by stefangordon** [comment]
I think this was fixed in #4584
